homeassistant/components/lgthinq/switch.py

```diff
@@ -0,0 +1,159 @@
+        ),
+    ),
+    ThinQProperty.BOILER_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=ThinQProperty.BOILER_OPERATION_MODE,
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=ThinQProperty.BOILER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+    ThinQProperty.DEHUMIDIFIER_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=ThinQProperty.DEHUMIDIFIER_OPERATION_MODE,
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=ThinQProperty.DEHUMIDIFIER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+    ThinQProperty.HUMIDIFIER_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=ThinQProperty.HUMIDIFIER_OPERATION_MODE,
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=ThinQProperty.HUMIDIFIER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+}
+
+DEVIE_TYPE_SWITCH_MAP: dict[DeviceType, tuple[ThinQSwitchEntityDescription, ...]] = {
+    DeviceType.AIR_PURIFIER_FAN: (
+        OPERATION_SWITCH_DESC[ThinQProperty.AIR_FAN_OPERATION_MODE],
+    ),
+    DeviceType.AIR_PURIFIER: (
+        OPERATION_SWITCH_DESC[ThinQProperty.AIR_PURIFIER_OPERATION_MODE],
+    ),
+    DeviceType.DEHUMIDIFIER: (
+        OPERATION_SWITCH_DESC[ThinQProperty.DEHUMIDIFIER_OPERATION_MODE],
+    ),
+    DeviceType.HUMIDIFIER: (
+        OPERATION_SWITCH_DESC[ThinQProperty.HUMIDIFIER_OPERATION_MODE],
+    ),
+    DeviceType.SYSTEM_BOILER: (
+        OPERATION_SWITCH_DESC[ThinQProperty.BOILER_OPERATION_MODE],
+    ),
+}
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ThinqConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up an entry for switch platform."""
+    for coordinator in entry.runtime_data.values():
+        descriptions = DEVIE_TYPE_SWITCH_MAP.get(coordinator.device_api.device_type)
+        if not isinstance(descriptions, tuple):
+            continue
+
+        entities: list[ThinQSwitchEntity] = []
+        for description in descriptions:
+            properties = create_properties(
+                device_api=coordinator.device_api,
+                key=description.key,
+                children_keys=None,
+                rw_type=PROPERTY_WRITABLE,
+            )
+            if not properties:
+                continue
+
+            entities += [
+                ThinQSwitchEntity(coordinator, description, prop) for prop in properties
+            ]
+
+        if entities:
+            for entity in entities:
+                _LOGGER.debug(
+                    "[%s] Add %s:%s entity",
+                    coordinator.device_name,
+                    Platform.SWITCH,
+                    entity.entity_description.key,
+                )
```

Comment:
This is already logged as far as I am aware

Reply:
Removed.